pinot-spi/src/main/java/org/apache/pinot/spi/accounting/ThreadAccountant.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.accounting;
+
+import javax.annotation.Nullable;
+
+
+public interface ThreadAccountant {
+
+  /**
+   * clear thread accounting info
+   */
+  void clear();
+
+  /**
+   * check if the corresponding runner thread of current thread is interrupted
+   */
+  boolean isRootThreadInterrupted();
+
+  /**
+   * Task tracking info
+   * @param queryId query id string
+   * @param taskId a unique task id
+   * @param parentContext the parent execution context, null for root(runner) thread
+   */
+  void createExecutionContext(String queryId, int taskId, @Nullable ExecutionContext parentContext);
+
+
+  ExecutionContext getExecutionContext();
+
+  /**
+   * set resource usage provider
+   */
+  void setThreadResourceUsageProvider(ThreadResourceUsageProvider threadResourceUsageProvider);
+
+  /**
+   * operator call to update thread cpu time
+   */
+  void sampleThreadCPUTime();
```

Comment:
Consider abstracting this out further. 

Probably accountant does not need to know or be told about what specific things to account. Since accountant relies on ResourceUsageProvider, the latter determines what resource usage will be accounted rather than caller of Accountant telling.

We can provide an abstract collect / account and based on the ResourceUsageProvider plugged into the Accountant, it will be called to sample the resource usage.

Reply:
The current approach is fine too. Just wondering if this can be made more flexible

Reply:
Good point, will abstract the sampling